src/Symfony/Component/VarDumper/Caster/DateCaster.php

```diff
@@ -31,7 +37,11 @@ public static function castDateTime(\DateTimeInterface $d, array $a, Stub $stub,
             .($location ? ($d->format('I') ? "\nDST On" : "\nDST Off") : '')
         ;
 
-        $a = [];
+        $a = array_filter($a, static function ($key) {
```

Comment:
no need for this: the while logic can be replaced by 3 unset, isn't it?

Reply:
Agreed, this should also fix the CI :)